### PR TITLE
button to validate the S3 Bucket configuration, update README

### PR DESCRIPTION
**Pull request by kuisathaverat**

### CODE REVIEW

**Comment by carlossg** [comment]
findbugs 

    Should io.jenkins.plugins.artifact_manager_jclouds.s3.S3BlobStoreConfig$S3BlobStoreTester be a _static_ inner class? 
    [io.jenkins.plugins.artifact_manager_jclouds.s3.S3BlobStoreConfig$S3BlobStoreTester] At S3BlobStoreConfig.java:[lines 80-90] SIC_INNER_SHOULD_BE_STATIC

File: src/main/resources/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig/config.jelly [hunk 1]
```diff
@@ -17,5 +17,7 @@
         <f:entry title="${%Delete Stashes}" field="deleteStashes">
             <f:checkbox readonly="true"/>
         </f:entry>
+        <f:validateButton title="Validate S3 Bucket configuration" progress="Validate..." method="validateS3BucketConfig"
```
Comment (jglick) [inline src/main/resources/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig/config.jelly]: Is this too expensive to just do automatically in a `doCheck*` method?

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java [hunk 2]
```diff
@@ -172,6 +194,35 @@ public FormValidation doCheckRegion(@QueryParameter String region){
         }
         return ret;
     }
-    
-    
+
+    public FormValidation doValidateS3BucketConfig(@QueryParameter String container, @QueryParameter String prefix,
+                                                   @QueryParameter String region){
+        FormValidation ret = FormValidation.ok("success");
```
Comment (jglick) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: You must check some permission here, probably `ADMINISTER`, and it should also be `@RequirePOST` to make sure there is no possibility of a SSRF or CSRF attack.

File: src/main/resources/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig/config.jelly (hunk 1 above)
Comment (kuisathaverat) [reply src/main/resources/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig/config.jelly]: it is expensive, it takes a few seconds.

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java [hunk 3]
```diff
@@ -172,6 +196,37 @@ public FormValidation doCheckRegion(@QueryParameter String region){
...
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        FormValidation ret = FormValidation.ok("success");
+        try {
+            S3BlobStore provider = new S3BlobStoreTester(container, prefix, region);
+            JCloudsVirtualFile jc = new JCloudsVirtualFile(provider, container, "");
+            jc.list();
+        } catch (Throwable t){
+            String msg = processExceptionMessage(t);
+            ret = FormValidation.error(StringUtils.abbreviate(msg, 200));
+            LOGGER.severe(t.getMessage());
```
Comment (jglick) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: Get rid of that. We do not want system log messages appearing if the admin has not actually _saved_ a bad config!

Comment (kuisathaverat) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: we can move it to FINEST for debug purposes

Comment (jglick) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: Sure.

**Pull request merged**